Conceptpower+Spring/src/main/java/edu/asu/conceptpower/rest/ConceptSearch.java

```diff
@@ -101,25 +107,20 @@ private void initBinder(WebDataBinder binder) {
 
         int page = 1;
 
+        preprocessConceptSearchParameterBean(conceptSearchParameters);
+
+        PropertyDescriptor pageDescriptor = new PropertyDescriptor("page", ConceptSearchParameters.class, "getPage",
+                "setPage");
+        if (pageDescriptor.getReadMethod().invoke(conceptSearchParameters, new Object[] {}) != null) {
+            page = Integer.parseInt(
+                    String.valueOf(pageDescriptor.getReadMethod().invoke(conceptSearchParameters, new Object[] {})));
+        }
+
         for (Field field : conceptSearchParameters.getClass().getDeclaredFields()) {
-            field.setAccessible(true);
-            if (field.getName().equalsIgnoreCase("type_uri")) {
-                if (field.get(conceptSearchParameters) != null) {
-                    searchFields.put("type_id",
-                            uriHelper.getTypeId(String.valueOf(field.get(conceptSearchParameters))));
-                }
-            } else if (SearchParamters.OPERATOR.equalsIgnoreCase(field.getName())) {
-                // If the value is null, then operator will be OR by default
-                if (field.get(conceptSearchParameters) != null) {
-                    operator = String.valueOf(field.get(conceptSearchParameters)).toUpperCase();
-                }
-            } else if (SearchParamters.PAGE.equalsIgnoreCase(field.getName())) {
-                page = field.get(conceptSearchParameters) != null ? (Integer) field.get(conceptSearchParameters) : page;
-            } else if (SearchParamters.NUMBER_OF_RECORDS_PER_PAGE.equalsIgnoreCase(field.getName())) {
-                numberOfRecordsPerPage = field.get(conceptSearchParameters) != null
-                        ? (Integer) field.get(conceptSearchParameters) : numberOfRecordsPerPage;
-            } else if (field.get(conceptSearchParameters) != null) {
-                searchFields.put(field.getName().trim(), String.valueOf(field.get(conceptSearchParameters)).trim());
+            PropertyDescriptor descriptor = new PropertyDescriptor(field.getName(), ConceptSearchParameters.class);
```

Comment:
where is the operator set? line 106 sets it to AND. line 127/8 pass it into the search.